### PR TITLE
Remove unreliable dependencies

### PR DESCRIPTION
**Pull request by hasufell**
See #1092 and #1089

### CODE REVIEW

**Comment by hasufell** [comment]
@jasagredo

**Comment by hasufell** [comment]
```
2024-07-05T18:58:17.2039026Z [1m[ Error ][0m []8;;https://errors.haskell.org/messages/GHCup-00200\GHCup-00200]8;;\] Digest error for /github/workspace/.ghcup/cache/ghc-8.10.3-armv7-deb10-linux.tar.xz: expected b823b58cae36fbac0741680ca7605180fa4cf4c6ae439123d282184b94d32fd6 but got a4ab25701d29a35dfd9124e48dfcb5850437a2b29e72fa36d22654b4d1b11a03 
```

Quite odd. The expected hash is indeed `b823b58cae36fbac0741680ca7605180fa4cf4c6ae439123d282184b94d32fd6` and I can't reproduce the error.

**Comment by Bodigrim** [comment]
> Quite odd. The expected hash is indeed `b823b58cae36fbac0741680ca7605180fa4cf4c6ae439123d282184b94d32fd6` and I can't reproduce the error.

Presumably it's https://github.com/GaloisInc/SHA/issues/16

**Comment by Bodigrim** [comment]
`SHA` package is likely to be magnitudes slower than `cryptohash-sha256`, I'd pretty much recommend not to change this.

File: lib/GHCup/CabalConfig.hs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    either (fail . show . fmap (showPError fp)) resolveConfig (parseConfig bs)
+
+-------------------------------------------------------------------------------
+-- Find config
+-------------------------------------------------------------------------------
+
+-- | Find the @~\/.cabal\/config@ file.
+findConfig :: IO FilePath
+findConfig = do
+    env <- lookupEnv "CABAL_CONFIG"
```
Comment (Bodigrim) [inline lib/GHCup/CabalConfig.hs:44]: For `cabal >= 3.12` you can query `cabal path | grep store-dir`.

**Pull request merged**